### PR TITLE
feat(web): next 15, react 19, proper caching

### PR DESCRIPTION
**Pull request by nahoc**
* proper redirects in app benchmarks
* proper error handling if something fails, won't crash the whole app
* caching set to 1 minute, uses ISR properly
* lighthouse 100 on all pages
* bump next to v15
* replace chartjs with recharts
* "Last updated" tooltip on datasheet page

### CODE REVIEW

**Review by morgangibbs87 — APPROVED** [review]

**Comment by flaub** [comment]
How long do you think before the rc/beta becomes released? I'd like to avoid using pre-released dependencies if possible.

**Review by flaub — CHANGES_REQUESTED** [review]
Holding off for now, hoping we can wait until these deps are stable.

**Comment by nahoc** [comment]
so Nextjs v15 is now in RC:
https://nextjs.org/blog/next-15-rc

I think it's safe to merge this PR now, which improves caching logic and other nits.

**Comment by nahoc** [comment]
also the email for key requested has `key` but the other one has `Key`, should be consistent 
image

**Review by pdg744 — APPROVED** [review]
If it meets Morgan's standards, LGTM :) 
Thank you!

**Review by flaub — CHANGES_REQUESTED** [review]
We should avoid using non-released software, especially for mission critical sites.

File: web/package.json [hunk 1]
```diff
@@ -12,37 +13,40 @@
...
+    "next": "15.0.0-rc.0",
     "next-themes": "0.3.0",
-    "prism-react-renderer": "2.3.1",
+    "prism-react-renderer": "2.4.0",
     "radash": "12.1.0",
-    "react": "18.3.1",
-    "react-chartjs-2": "2.11.2",
-    "react-dom": "18.3.1",
+    "react": "19.0.0-rc-e210d081-20240909",
+    "react-dom": "19.0.0-rc-e210d081-20240909",
```
[the remark this answers is not in the document]
Comment (nahoc) [reply web/package.json]: for Next15 to work, yes
`$ unhandledRejection Error: Next.js requires react >= 19.0.0 to be installed.`
react 19 should be released very soon, but no clear date :/

**Comment by nahoc** [comment]
@flaub this PR has been open for around 5 months now, do you think it's stable enough to merge? I got a few other improvements depending on it. 🙏

**Comment by flaub** [comment]
If we can drop the `-rc` dependencies then I'll approve.

**Comment by nahoc** [comment]
> If we can drop the `-rc` dependencies then I'll approve.

OK, just downgraded to React v18 for the time being. 👍

**Review by flaub — APPROVED** [review]

**Pull request merged**